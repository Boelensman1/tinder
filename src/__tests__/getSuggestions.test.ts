import * as nock from 'nock';

import TinderClient from '..'; // tslint:disable-line import-name
import { token, id } from './util/getAuth';

nock.load('./src/__tests__/mocks/auth.json');
nock.load('./src/__tests__/mocks/getSuggestions.json');

const tinderClient = new TinderClient();

beforeAll(() => (
  tinderClient.auth(token, id)
));

it('gets recommentations', async () => {
  const suggestions = await tinderClient.getSuggestions();
  expect(suggestions).toBeDefined();
  expect(suggestions[0].birthDate).toBeInstanceOf(Date);

  expect(suggestions).toMatchSnapshot();
});
