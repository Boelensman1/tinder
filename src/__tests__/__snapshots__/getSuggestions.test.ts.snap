// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets recommentations 1`] = `
Array [
  Profile {
    "bio": "aaaaaaaaaa",
    "birthDate": 1995-01-01T08:11:14.795Z,
    "birthDateInfo": "fuzzy birthdate active, not displaying real birth_date",
    "commonConnections": Array [],
    "commonInterests": Array [],
    "connectionCount": 0,
    "contentHash": "c1639a04bab9114a3945f5219a88c274f6ac38de45jhqau",
    "distanceMi": 20,
    "groupMatched": false,
    "id": "51d4cc502e7211e8a74c61df",
    "instagram": undefined,
    "isFemale": true,
    "isTraveling": undefined,
    "jobs": Array [
      Object {
        "company": Object {
          "name": "jobname",
        },
        "title": Object {
          "name": "jobtitle",
        },
      },
    ],
    "name": "Suggestion1",
    "photos": Array [
      Photo {
        "extension": "jpg",
        "fileName": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
        "processedFiles": Array [
          Object {
            "height": 640,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/640x640_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 640,
          },
          Object {
            "height": 320,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/320x320_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 320,
          },
          Object {
            "height": 172,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/172x172_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 172,
          },
          Object {
            "height": 84,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/84x84_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 84,
          },
        ],
        "tinderClient": TinderClient {
          "http": HttpClient {
            "authToken": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
            "axiosClient": [Function],
          },
        },
        "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
      },
    ],
    "pingTime": 2014-12-09T00:00:00.000Z,
    "sNumber": 100000000,
    "schools": Array [],
    "spotifyThemeTrack": undefined,
    "teaser": Object {
      "string": "position",
      "type": "jobPosition",
    },
    "teasers": Array [
      Object {
        "string": "position",
        "type": "jobPosition",
      },
    ],
    "tinderClient": TinderClient {
      "http": HttpClient {
        "authToken": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
        "axiosClient": [Function],
      },
    },
  },
  Profile {
    "bio": "bbbbbbbbbb",
    "birthDate": 1997-01-01T08:00:00.000Z,
    "birthDateInfo": "fuzzy birthdate active, not displaying real birth_date",
    "commonConnections": Array [],
    "commonInterests": Array [],
    "connectionCount": 0,
    "contentHash": "c1639a04bab9114a3945f5219a88c274f6ac38de4",
    "distanceMi": 50,
    "groupMatched": false,
    "id": "52d4cc502e7211e8a74c61df",
    "instagram": undefined,
    "isFemale": true,
    "isTraveling": undefined,
    "jobs": Array [
      Object {
        "company": Object {
          "name": "name",
        },
      },
    ],
    "name": "Suggestion2",
    "photos": Array [
      Photo {
        "extension": "jpg",
        "fileName": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
        "processedFiles": Array [
          Object {
            "height": 640,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/640x640_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 640,
          },
          Object {
            "height": 320,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/320x320_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 320,
          },
          Object {
            "height": 172,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/172x172_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 172,
          },
          Object {
            "height": 84,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/84x84_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 84,
          },
        ],
        "tinderClient": TinderClient {
          "http": HttpClient {
            "authToken": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
            "axiosClient": [Function],
          },
        },
        "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
      },
    ],
    "pingTime": 2014-12-09T00:00:00.000Z,
    "sNumber": 200000000,
    "schools": Array [
      Object {
        "id": "111111111111111",
        "name": "schoolname",
      },
    ],
    "spotifyThemeTrack": undefined,
    "teaser": Object {
      "string": "positionname",
      "type": "position",
    },
    "teasers": Array [
      Object {
        "string": "positionname",
        "type": "position",
      },
      Object {
        "string": "schoolname",
        "type": "school",
      },
    ],
    "tinderClient": TinderClient {
      "http": HttpClient {
        "authToken": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
        "axiosClient": [Function],
      },
    },
  },
]
`;

exports[`likes a recommentations 1`] = `
Object {
  "isMatch": true,
  "likesRemaining": 100,
}
`;

exports[`superLikes a recommentations 1`] = `
Object {
  "isMatch": false,
  "superLikes": Object {
    "alc_remaining": 0,
    "allotment": 1,
    "new_alc_remaining": 0,
    "remaining": 0,
    "resets_at": "2018-01-01T01:01:01.001Z",
    "superlike_refresh_amount": 5,
    "superlike_refresh_interval": 1,
    "superlike_refresh_interval_unit": "d",
  },
}
`;
